File: src/cs/Ssh/Events/SshSessionClosedEventArgs.cs
// Copyright (c) Microsoft. All rights reserved.

using System;
using System.Diagnostics;

namespace Microsoft.DevTunnels.Ssh.Events;

[DebuggerStepThrough]
public class SshSessionClosedEventArgs : EventArgs
{
	public SshSessionClosedEventArgs(SshDisconnectReason reason, string message, Exception? ex)
	{
		Reason = reason;
		Message = message;
		Exception = ex;
	}

	public SshDisconnectReason Reason { get; }

	public string Message { get; }

	public Exception? Exception { get; }
}

File: src/ts/ssh/events/sshSessionClosedEventArgs.ts
//
//  Copyright (c) Microsoft Corporation. All rights reserved.
//

import { SshDisconnectReason } from '../messages/transportMessages';

export class SshSessionClosedEventArgs {
	public constructor(
		public readonly reason: SshDisconnectReason,
		public readonly message: string,
		public readonly error: Error | null,
	) {
		if (!message) throw new TypeError('A disconnect message is required.');
	}

	public toString() {
		return `${SshDisconnectReason[this.reason]}: ${this.message}`;
	}
}
